### PR TITLE
[13.0] ADD web chatt filtering functionality

### PR DESCRIPTION
**Pull request by pegonzalezspesol**
After working with Odoo a long time and load lot of loaded data, the amount of messages displayed inside the chatter of a record start to be significantly big

This module introduces a filter in the Chatter and next functionalities are provided:

Filter messages by content
Filter messages by author
Reverse filtering on author
F.ex. <-admin> will show all the messages in the chatt excluding the ones of the Administrator

### CODE REVIEW

File: web_chatt_filtering/__init__.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
+##############################################################################
```
Comment (tarteo) [inline web_chatt_filtering/__init__.py]: Please use a smaller copyright flag. https://github.com/OCA/maintainer-tools/blob/master/template/module/__manifest__.py

File: web_chatt_filtering/__manifest__.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    "name": "mail_chatt_filtering",
+    "summary": """
+        Extend the mail_chatt functionality to have filtering by author or by
+         message content
+    """,
+    "category": "web",
+    "author": "Pedro Gonzalez <pedro.gonzalez@pesol.es> ,"
+    "Odoo Community Association (OCA)",
+    "license": "AGPL-3",
+    "website": "http://github.com/OCA/web",
```
Comment (tarteo) [inline web_chatt_filtering/__manifest__.py]: https, please.

File: web_chatt_filtering/static/src/js/chatter.js [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    Chatt.include({
+        events: _.extend({}, Chatt.prototype.events, {
+            'keyup input#search_partner': function (ev) {
+                if (ev.which === $.ui.keyCode.ENTER) {
+                    this.filter_message($(ev.target).val().trim());
+                }
+            },
+        }),
+
+        filter_message: function (substring) {
```
Comment (tarteo) [inline web_chatt_filtering/static/src/js/chatter.js]: Can you refactor this function so it's easier inheritable. E.g. put the rpc call inside a separate function.

**Review by tarteo — CHANGES_REQUESTED** [review]
Cool module!
I have some small remarks:
* When I search and there's 0 messages found it shows every message. Can you make it so something like "No messages found" appears.

**Review by tarteo — COMMENTED** [review]
The no messages found is now put into the search input box. Is this as intended? Now it seems there are conflicts can you fix those, please.

**Review by tarteo — APPROVED** [review]
Thanks!

**Review by tarteo — APPROVED** [review]
Thanks!

**Comment by pegonzalezspesol** [comment]
> Thanks!

Thanks for your revision Dennis! @tarteo

**Comment by pegonzalezspesol** [comment]
@hbrunn, @daramousk can any of you please review this PR?

**Comment by pegonzalezspesol** [comment]
@angelmoya could you please review?

**Pull request closed without merging**